### ISSUE TITLE
Compatibility with Nuxt 3 ?

### ISSUE THREAD

**Issue by copostic** [vue-youtube/vue-youtube#4]
Hi there,

I've been struggling for hours trying to make this great module to work on my Nuxt 3 project but I'm beggining to think that it's just not compatible...

In fact, the plugin works well, but once I trigger a navigation by clicking on a Nuxt link and coming back to the page that contains de Youtube video, it's not loaded anymore.

I tried to add the usePlayer on the onBeforeMount hook and use clearVideo on the onBeforeUnmount, to set the videoId to null and reset it on mount but nothing works...

When the page is reloaded, the id of the div that I pass as ref is incremented (id="vue-youtube-2", id="vue-youtube-3"...etc)  but the onReady is never executed neither the onError. 

Any idea of what I could do to make it work on Nuxt ?

Thanks a lot :)

**Comment by Techassi** [comment]
Do you have a minimal reproduction case, maybe as a Git repo or a StackBlitz link? This makes it easier for me too look into this issue.

**Comment by copostic** [comment]
@Techassi Yep of course, sorry, here's the Stackblitz with the MRE : [MRE Nuxt 3 - vue-youtube](https://stackblitz.com/edit/nuxt-starter-vvjchq)

As you can see on the first load, the Youtube video is on the page but when you click on "Go to test" and then back on "Go to index", the video will not be build...

Thanks for your time ! :)

**Comment by Techassi** [comment]
Great, thank you! I will look into it. I guess it's some kind of issue with state management of the player manager.

I'm planning to release a special Nuxt package for further Nuxt integration. I will keep you updated!

**Comment by Techassi** [comment]
It is indeed related to the state management. I will release a Nuxt module `@vue-youtube/nuxt` with the `0.0.3` version. It will correctly create the manager and register it to the Nuxt (Vue) application. The module can then be used like:

```ts
export default defineNuxtConfig({
  modules: [
    '@vue-youtube/nuxt'
  ]
})
```

See [here](https://v3.nuxtjs.org/guide/features/modules) for more information on Nuxt modules.

**Comment by copostic** [comment]
Okey dokey great thank you, if I can be of any help to test it for example don't hesitate 😁

**Comment by Techassi** [comment]
I implemented Nuxt support, which should work in theory but yet it doesn't. Seems like a Nuxt issue... I wanted to release Nuxt support with the 0.0.3, but it seems like it will be pushed back to the 0.0.4 release.

But I can still provide the (in theory) working code on a seperate branch if you would like to ticker with it.

**Comment by CodyBontecou** [comment]
I'm happy to take a look if you post the branch.

**Comment by Techassi** [comment]
Great will push the changes soon.

**Comment by Techassi** [comment]
Published the (non working) Nuxt module on the [`nuxt`](https://github.com/vue-youtube/vue-youtube/tree/nuxt) branch. Maybe one will spot the mistake I made...

**EDIT**

Add mention @CodyBontecou

**Comment by CodyBontecou** [comment]
Cool, I'll see if I can help

**Comment by CodyBontecou** [comment]
I'm not sure why the latest release doesn't work with Nuxt 3, but I got this to work using v0.0.2

**Comment by Techassi** [comment]
Yeah this is an issue with inject/provide. The manager is provided via the plugin setup hook:

```typescript
install(app) {
  app.provide(PROVIDE_KEY, manager);
  this._insert(); // Insert the needed script tags, unrelated to this issue
},
```

> See [here](https://github.com/vue-youtube/vue-youtube/blob/main/packages/core/manager.ts#L58)

I'm still not quite sure why Nuxt v3 has these issues. I have some code locally which I did not push to the `nuxt` branch. I can provide it if needed.

Also there is an issue with Monorepos and Nuxt, as it cannot find some of the dependencies: `500 Cannot read properties of undefined (reading 'modules')`. This is a known [issue](https://github.com/nuxt/nuxt/issues/14952).
